File: app/src/main/java/com/devusercode/upchat/adapter/UserAdapter.java
package com.devusercode.upchat.adapter;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.devusercode.upchat.ConversationActivity;
import com.devusercode.upchat.R;
import com.devusercode.upchat.models.User;
import com.firebase.ui.database.FirebaseRecyclerAdapter;
import com.firebase.ui.database.FirebaseRecyclerOptions;
import com.google.android.material.card.MaterialCardView;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class UserAdapter extends FirebaseRecyclerAdapter<User, UserAdapter.UserViewHolder> {
    private final String TAG = this.getClass().getSimpleName();
    private final Context currentContext;
    private final FirebaseUser fuser = FirebaseAuth.getInstance().getCurrentUser();

    public UserAdapter(@NonNull Context context, @NonNull FirebaseRecyclerOptions<User> options) {
        super(options);
        this.currentContext = context;
    }

    @Override
    protected void onBindViewHolder(@NonNull UserViewHolder holder, int position, @NonNull User user) {
        // Bind user data to the view holder
        if (!fuser.getUid().equals(user.getUid())) {
            holder.username.setText(user.getUsername());
            holder.email.setText(user.getEmail());

            // Load a profile picture into the view holder
            if (!user.getPhotoUrl().isEmpty()) {
                Glide.with(currentContext).load(Uri.parse(user.getPhotoUrl())).placeholder(R.drawable.ic_account_circle_black).circleCrop().into(holder.profile_image);
            } else {
                holder.profile_image.setImageResource(R.drawable.ic_account_circle_black);
            }
        } else {
            // Exclude your own user
            holder.root.setVisibility(View.GONE);
        }

        // Handle add user button click
        holder.add_user_button.setOnClickListener(view -> {
            Intent intent = new Intent(currentContext, ConversationActivity.class);
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            intent.putExtra("uid", user.getUid());
            currentContext.startActivity(intent);
        });
    }

    @NonNull
    @Override
    public UserViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.item_user, parent, false);
        return new UserViewHolder(view);
    }

    static class UserViewHolder extends RecyclerView.ViewHolder {
        TextView username, email;
        ImageView profile_image;
        LinearLayout root;
        MaterialCardView materialcardview1;
        Button add_user_button;

        public UserViewHolder(@NonNull View view) {
            super(view);

            root = view.findViewById(R.id.linear1);
            materialcardview1 = view.findViewById(R.id.materialcardview1);
            profile_image = view.findViewById(R.id.profile_image);
            username = view.findViewById(R.id.username);
            email = view.findViewById(R.id.email);
            add_user_button = view.findViewById(R.id.add_user_button);
        }
    }
}

File: app/src/main/java/com/devusercode/upchat/adapter/UserAdapter.kt
package com.devusercode.upchat.adapter

import android.content.Intent
import android.net.Uri
import android.os.Build
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.devusercode.upchat.ConversationActivity
import com.devusercode.upchat.R
import com.devusercode.upchat.UserProfileActivity
import com.devusercode.upchat.models.User
import com.firebase.ui.database.FirebaseRecyclerAdapter
import com.firebase.ui.database.FirebaseRecyclerOptions
import com.google.android.material.card.MaterialCardView
import com.google.firebase.auth.FirebaseAuth

class UserAdapter(app: AppCompatActivity, options: FirebaseRecyclerOptions<User?>) :
    FirebaseRecyclerAdapter<User, UserAdapter.UserViewHolder>(options) {
    private val TAG = this.javaClass.simpleName
    private val fuser = FirebaseAuth.getInstance().currentUser
    private val context = app.applicationContext

    @RequiresApi(Build.VERSION_CODES.O)
    override fun onBindViewHolder(holder: UserViewHolder, position: Int, user: User) {
        // Exclude yourself from the list
        if (fuser != null && user.uid == fuser.uid) {
            holder.root.visibility = View.GONE
            return
        }

        // Bind user data to the view holder
        holder.username.text = user.username
        holder.email.text = user.email

        // Load a profile picture into the view holder
        if (user.photoUrl!!.isNotEmpty()) {
            Glide.with(context).load(Uri.parse(user.photoUrl))
                .placeholder(R.drawable.ic_account_circle_black).circleCrop()
                .into(holder.profile_image)
        } else {
            holder.profile_image.setImageResource(R.drawable.ic_account_circle_black)
        }


        // Handle add user button click
        holder.add_user_button.setOnClickListener {
            val intent = Intent(context, ConversationActivity::class.java)
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            intent.putExtra("uid", user.uid)
            context.startActivity(intent)
        }

        holder.materialcardview1.setOnClickListener {
            val intent = Intent(context, UserProfileActivity::class.java)
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            intent.putExtra("uid", user.uid)
            context.startActivity(intent)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.item_user, parent, false)
        return UserViewHolder(view)
    }

    class UserViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var username: TextView
        var email: TextView
        var profile_image: ImageView
        var root: LinearLayout
        var materialcardview1: MaterialCardView
        var add_user_button: Button

        init {
            root = view.findViewById(R.id.linear1)
            materialcardview1 = view.findViewById(R.id.materialcardview1)
            profile_image = view.findViewById(R.id.profile_image)
            username = view.findViewById(R.id.username)
            email = view.findViewById(R.id.email)
            add_user_button = view.findViewById(R.id.add_user_button)
        }
    }
}